api/v1alpha1/dataplane_konnect_extension_types.go

```diff
@@ -78,15 +79,28 @@ type DataPlaneKonnectExtensionSpec struct {
 	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
 	ServerHostname string `json:"serverHostname"`
 
-	// ClusterCertificateSecretName is a name of the Secret containing the Konnect Control Plane's cluster certificate.
+	// KonnectAuthConfiguration must be used to configure the Konnect API authentication.
 	// +kubebuilder:validation:Required
-	ClusterCertificateSecretName string `json:"clusterCertificateSecretName"`
+	KonnectAuthConfiguration KonnectAPIAuthConfiguration `json:"konnectAuthConfiguration"`
 
 	// ClusterDataPlaneLabels is a set of labels that will be applied to the Konnect DataPlane.
 	// +optional
 	ClusterDataPlaneLabels map[string]string `json:"clusterDataPlaneLabels,omitempty"`
 }
 
+// KonnectAPIAuthConfiguration contains the configuration for the Konnect API authentication.
+type KonnectAPIAuthConfiguration struct {
```

Comment:
I think it would make sense to name it more precisely, i.e. `KonnectControlPlaneAPIAuthConfiguration`. Reasons:
1. It's used to authenticate DPs with Control Plane API
2. We already have `KonnectAPIAuthConfiguration` in `kubernetes-configuration`

Reply:
it makes sense. Changed in 660b0c7e67cef5a87505110e6383500fb42bf921